install/0000_00_cluster-version-operator_00_namespace.yaml

```diff
@@ -10,3 +10,6 @@ metadata:
     name: openshift-cluster-version
     openshift.io/run-level: "1"
     openshift.io/cluster-monitoring: "true"
+    pod-security.kubernetes.io/enforce: privileged
+    pod-security.kubernetes.io/audit: privileged
+    pod-security.kubernetes.io/warn: privileged
```

Comment:
It's not clear to me why the CVO would need to be `privileged`.  The CVO isn't trying to do anything particularly fancy at the container-security level, and it would be nice if we could fit ourselves into the [`restricted` policy][1].  We do have [two `hostPath` mounts][2] today, but one of those if for mounting TLS certs, and surely there is some way to get at those without requiring all the access `privileged` gets us?  I expect we could also do something similar with volumes for `etc-cvo-updatepayloads` to pass payloads from one pod to future pods.  Is there some discussion somewhere about collaborating on changes needed to move to `restricted`?  At least the CA-mounting thing seem like they'll be common problems.

[1]: https://kubernetes.io/docs/concepts/security/pod-security-standards/#restricted
[2]: https://github.com/openshift/cluster-version-operator/blob/e816c118ac608f131d24b28d617e91d9d5cc34a6/install/0000_00_cluster-version-operator_03_deployment.yaml#L90-L95

Reply:
@wking the concrete violation is as follows:
```
$ jq  '. | select((.objectRef.resource == "deployments") and (.annotations | has("pod-security.kubernetes.io/audit")) and (.objectRef.namespace == "openshift-cluster-version")) | {name: .objectRef.name, resource: .objectRef.resource, podSecurityReason: .annotations["pod-security.kubernetes.io/audit"] }' audit.log | jq -C -s --sort-keys '. | unique'
[
  {
    "name": "cluster-version-operator",
    "podSecurityReason": "host namespaces (hostNetwork=true), hostPath volumes (volumes \"etc-ssl-certs\", \"etc-cvo-updatepayloads\")",
    "resource": "deployments"
  }
]
```

Reply:
so you are declaring
- host networking
- host paths
which violates baseline policy

Reply:
if you can get rid of those you'd at least baseline policy.

Reply:
Note: there is no level between baseline and restricted.

Reply:
>  surely there is some way to get at those without requiring all the access privileged gets us?

unfortnately, with the current Pod Security admission, no.